MvvmCross/Droid/Droid/Views/MvxAndroidViewPresenter.cs

```diff
@@ -40,8 +40,19 @@ protected virtual void Show(Intent intent)
 
         protected virtual Intent CreateIntentForRequest(MvxViewModelRequest request)
         {
-            var requestTranslator = Mvx.Resolve<IMvxAndroidViewModelRequestTranslator>();
-            var intent = requestTranslator.GetIntentFor(request);
+            IMvxAndroidViewModelRequestTranslator requestTranslator = Mvx.Resolve<IMvxAndroidViewModelRequestTranslator>();
+            Intent intent;
+
+            if (request is MvxViewModelInstanceRequest)
+            {
+                var instanceRequest = requestTranslator.GetIntentWithKeyFor(((MvxViewModelInstanceRequest)request).ViewModelInstance);
+                intent = instanceRequest.Item1;
+            }
+            else
+            {
+                intent = requestTranslator.GetIntentFor(request);
+            }
+
```

Comment:
Again remove the `intent` variable in the outer scope and change all this to an early return as in `if () {... return ...} return ....`